Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1587,69 +1582,89 @@ RcTcacheEntry TextureCacheBase::GetTexture(const int textureCacheSafetyColorSamp
     InvalidateTexture(oldest_entry);
   }
 
-  std::shared_ptr<HiresTexture> hires_tex;
+  std::optional<VideoCommon::CustomTextureData> data;
+  bool has_arbitrary_mipmaps = false;
   if (g_ActiveConfig.bHiresTextures)
   {
-    hires_tex = HiresTexture::Search(texture_info);
-
-    if (hires_tex)
+    auto hires_texture = HiresTexture::Search(texture_info);
+    if (hires_texture)
     {
-      const auto& level = hires_tex->GetData().m_levels[0];
-      if (level.width != width || level.height != height)
-      {
-        width = level.width;
-        height = level.height;
-      }
-      expanded_width = level.width;
-      expanded_height = level.height;
+      data = hires_texture->GetData();
```

Comment:
This invokes a copy on the `CustomTextureData` I believe, which looks potentially expensive. You probably want a different way to handle this.

Reply:
Thanks Admiral.  This bit will be changing a bit in my next PR and then probably even more in a future PR once I have all the other stuff worked out.  I pushed up a stop-gap here, tested and I think it avoids the copy.  Great catch.

Reply:
Ah, nm, that's still a copy...probably need a reference :(

Reply:
Ok, sorry for the initial dumbness.  Should be good now :D